moonstone/plot/plot_template.py

```diff
@@ -0,0 +1,115 @@
+import math
+import numpy as np
+import pandas as pd
+import plotly.graph_objects as go
+import plotly.io
+
+
+class BaseGraph():
```

Comment:
We could define a abstract class here too.

Here is an example : https://github.com/motleystate/moonstone/pull/10/files#diff-7492a568bb2304da420600cf5b0b3d7eR8

A bit of documentation about it too: https://www.geeksforgeeks.org/abstract-classes-in-python/